### PR TITLE
TST: Test for the Enum triggering TypeError (#22551 issue)

### LINKED ISSUE #22551: columns comparison with Enum column name triggers TypeError
#### Code Sample, a copy-pastable example if possible

```python
from enum import Enum
import pandas as pd

Cols = Enum('Cols', 'col1 col2')

q1 = pd.DataFrame({Cols.col1: [1, 2, 3]})
q2 = pd.DataFrame({Cols.col1: [1, 2, 3]})

print((q1[Cols.col1] == q2[Cols.col1]).all())
```
#### Problem description

On pandas 0.23.x, comparison of columns in DataFrames which have enum.Enum column names triggers off a TypeError exception. it looks that pandas calls user defined Enum class as a function. 

```
Traceback (most recent call last):
  File "./pandas_enum.py", line 9, in <module>
    print((q1[Cols.col1] == q2[Cols.col1]).all())
  File "/.../lib/python3.6/site-packages/pandas/core/ops.py", line 1217, in wrapper
    name=res_name).rename(res_name)
  File "/.../lib/python3.6/site-packages/pandas/core/series.py", line 3318, in rename
    return super(Series, self).rename(index=index, **kwargs)
  File "/.../lib/python3.6/site-packages/pandas/core/generic.py", line 973, in rename
    level=level)
  File "/.../lib/python3.6/site-packages/pandas/core/internals.py", line 3340, in rename_axis
    obj.set_axis(axis, _transform_index(self.axes[axis], mapper, level))
  File "/.../lib/python3.6/site-packages/pandas/core/internals.py", line 5298, in _transform_index
    items = [func(x) for x in index]
  File "/.../lib/python3.6/site-packages/pandas/core/internals.py", line 5298, in <listcomp>
    items = [func(x) for x in index]
TypeError: 'Cols' object is not callable
```

#### Expected Output

just print `True`. Pandas 0.22.0 does the expected behavior. Pandas 0.23.0 and later has this problem. 

#### Output of ``pd.show_versions()``

<details>

```
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Linux
OS-release: 3.10.0-327.4.5.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: None
pip: 18.0
setuptools: 40.2.0
Cython: None
numpy: 1.15.1
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
```

</details>

### PR DESCRIPTION
**Pull request by Shadimrad**
- [x] closes #22551

### CODE REVIEW

File: pandas/tests/frame/indexing/test_get_value.py [hunk 1]
```diff
@@ -20,3 +22,16 @@ def test_get_value(self, float_frame):
                 result = float_frame._get_value(idx, col)
                 expected = float_frame[col][idx]
                 assert result == expected
+
+
+def test_enum_value():
```
Comment (mroeschke) [inline pandas/tests/frame/indexing/test_get_value.py:27]: 1. Could you name this test `test_enum_column_equality`
2. Could you move this to `pandas/tests/frame/test_arithmetic.py`?
3. Also below could you structure the test like
```
result = q1[Cols.col1] == q2[Cols.col1]
expected = ...
tm.assert_equal(result, expected)
```

Comment (Shadimrad) [reply pandas/tests/frame/indexing/test_get_value.py:27]: Sure! Thank you for your guide.

Comment (Shadimrad) [reply pandas/tests/frame/indexing/test_get_value.py:27]: @mroeschke do I remove the `.all()` method as well?

Comment (mroeschke) [reply pandas/tests/frame/indexing/test_get_value.py:27]: Correct

Comment (mroeschke) [reply pandas/tests/frame/indexing/test_get_value.py:27]: The changes in this file can be removed now.

Comment (Shadimrad) [reply pandas/tests/frame/indexing/test_get_value.py]: Sure. About removing `a.all()` when I create a Pandas.Series with the same truth values to compare with the _result_, it tells me that the truth value of a series is ambiguous and I should either of `a.empty, a.bool(), a.item(), a.any() or a.all().`

Comment (mroeschke) [reply pandas/tests/frame/indexing/test_get_value.py]: If you inspect `result`, it's a Series

```
In [3]: result
Out[3]:
0    True
1    True
2    True
Name: Cols.col1, dtype: bool
```

So `expected` should also be a `Series`

Comment (Shadimrad) [reply pandas/tests/frame/indexing/test_get_value.py]: Right. but when I created the series and tried to compare, 

```
data = np.array([True, True, True])
expected = pd.Series(data)
```
checking if `expected == result` gave me the Error that the truth value of the Series is ambiguous, that is why. Unless `tm.assert_equal` accounts for that.

Comment (mroeschke) [reply pandas/tests/frame/indexing/test_get_value.py]: Yup, `tm.assert_equal` will check the values and the aspects of the Series

```
In [1]:  from enum import Enum

In [2]:     Cols = Enum("Cols", "col1 col2")
   ...:
   ...:     q1 = DataFrame({Cols.col1: [1, 2, 3]})
   ...:     q2 = DataFrame({Cols.col1: [1, 2, 3]})
   ...:
   ...:     result = q1[Cols.col1] == q2[Cols.col1]

In [3]: result
Out[3]:
0    True
1    True
2    True
Name: Cols.col1, dtype: bool

In [4]: import pandas._testing as tm

In [5]: epxected = pd.Series([True, True, True], name=Cols.col1)

In [6]: tm.assert_series_equal(result, epxected)
```

Comment (Shadimrad) [reply pandas/tests/frame/indexing/test_get_value.py]: Update: changing it to `tm.assert_series_equal(result, expected)` , hoping it works.

Comment (Shadimrad) [reply pandas/tests/frame/indexing/test_get_value.py]: Oh, I've just seen your reply. There was a syntax issue that I fixed. `pd.Series` now replaced with `Series` which hopefully should be working now.

**Comment by Shadimrad** [comment]
I believe it's done! @mroeschke

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks, great job @Shadimrad!

**Pull request merged**